### DISCUSSION TITLE
Prometheus queries (queued jobs) for Actions Metrics Server

### DISCUSSION THREAD

**Discussion by duhow** [actions/actions-runner-controller#2631] [Questions]
I am writing some Prometheus queries to create Grafana dashboards, but I'm losing my head with the incoming data...

Apparently, there is the case when (for example):
- There are 40 GHA jobs currently running
- Actions Metrics Server starts
- Metrics add +40 jobs `completed`
- There are 0 jobs `started` as the webhook call was not received/processed.

When running the following query, this causes that "base line" is **over 0**.
```sql
sum(sum_over_time(github_workflow_jobs_queued_total{runs_on=~"(.+)?runner-group(.+)?"}[60s])) - 
sum(sum_over_time(github_workflow_jobs_started_total{runs_on=~"(.+)?runner-group(.+)?"}[60s]))
```

![image](https://github.com/actions/actions-runner-controller/assets/1145001/4a2809db-7aa8-41d9-8254-603311defbea)

In order to fix this, I have to **manually substract** a variable amount (`330` in this case) to fix this visualization:

```sql
clamp_min(
    sum(sum_over_time(github_workflow_jobs_queued_total{runs_on=~"(.+)?runner-group(.+)?"}[60s])) - 
    sum(sum_over_time(github_workflow_jobs_started_total{runs_on=~"(.+)?runner-group(.+)?"}[60s])) - 330
, 0)
```

![image](https://github.com/actions/actions-runner-controller/assets/1145001/69c4f2ce-ad68-4935-9588-34cc2ebb7b88)

Any idea on how to fix this numbers or get the minimum value over time to substract automatically?

**Comment by Samsanga** [comment]
@duhow , Did you manage to get `actions-metrics-server` working on GitHub Enterprise or GitHub.com ?

**Comment by duhow** [comment]
Yes, we are using GitHub Cloud (GitHub Team).
